appeals/web/src/server/lib/display-page-formatter.js

```diff
@@ -0,0 +1,119 @@
+import logger from '#lib/logger.js';
+import { buildHtmSpan } from '#lib/nunjucks-template-builders/tag-builders.js';
+import { appealShortReference } from './nunjucks-filters/appeals.js';
+export const formatDocumentActionLink = (
+	/** @type {any} */ appealId,
+	/** @type {any} */ listOfDocuments
+) => {
+	// If list of documents is in the correct format
+	if (
+		listOfDocuments != null &&
+		typeof listOfDocuments === 'object' &&
+		'documents' in listOfDocuments
+	) {
+		if (Array.isArray(listOfDocuments.documents) && listOfDocuments.documents.length > 0) {
+			const document = listOfDocuments.documents[0];
+			return `/appeals-service/appeal-details/${document.caseId}/documents/${document.folderId}/upload/${document.id}`;
+		}
+		return `/appeals-service/appeal-details/${appealId}/documents/${listOfDocuments.folderId}/upload/`;
+	}
+	return `#`;
+};
+
+export const formatListOfNotificationMethodsToHtml = (
+	/** @type {string | any[]} */ notificationMethods
+) => {
+	let html = ``;
+	if (notificationMethods.length > 0) {
+		for (const method of notificationMethods) {
+			html += `<span>${method.name}</span>`;
+			html += notificationMethods.indexOf(method) !== notificationMethods.length - 1 ? `</br>` : ``;
+		}
+	}
+	return html;
+};
+
+/**
+ * @param {string} answer
+ * @param {string} details
+ */
+export const formatAnswerAndDetails = (answer, details) => {
+	return answer === 'Yes'
+		? `${buildHtmSpan(answer)}<br>${buildHtmSpan(details)}`
+		: `${buildHtmSpan(answer)}`;
+};
+
+export const formatListOfAppeals = (
+	/** @type {{appealId: Number, appealReference: string}[]} */ listOfAppeals
+) => {
+	if (listOfAppeals.length > 0) {
+		let formattedLinks = ``;
+		for (let i = 0; i < listOfAppeals.length; i++) {
+			const shortAppealReference = appealShortReference(listOfAppeals[i].appealReference);
+			formattedLinks += `<a href='/appeals-service/appeal-details/${listOfAppeals[i].appealId}' class="govuk-link">${shortAppealReference}</a>`;
+			formattedLinks += i + 1 >= listOfAppeals.length ? '' : '<br>';
+		}
+		return `<span>${formattedLinks}</span>`;
+	}
+	return undefined;
+};
+
+/** @returns {string} */
+export const formatListOfListedBuildingNumbers = (
+	/** @type {string | any[]} */ listOfListedBuildingNumbers
+) => {
+	if (listOfListedBuildingNumbers.length > 0) {
+		let formattedLinks = ``;
+		for (let i = 0; i < listOfListedBuildingNumbers.length; i++) {
+			const listedBuildingNumber = listOfListedBuildingNumbers[i];
+			if ('listEntry' in listedBuildingNumber) {
+				formattedLinks += `<a href='https://historicengland.org.uk/listing/the-list/list-entry/${listedBuildingNumber.listEntry}' class="govuk-link">${listedBuildingNumber.listEntry}</a>`;
+				formattedLinks += i + 1 >= listOfListedBuildingNumbers.length ? '' : '<br>';
+			} else {
+				formattedLinks += `<a href='https://historicengland.org.uk} class="govuk-link"'>ListEntryNumber not yet in BD</a>`;
```

Comment:
There's a typo here in the `href` attribute which is causing the class to be rendered as part of the href